Modules/SubModule_Import/xASL_bids_BIDSifyCheckTimeEncoded.m

```diff
@@ -62,14 +62,26 @@
         bTimeEncodedFME = ~isempty(regexp(jsonOut.SeriesDescription,'(Encoded_Images_Had)\d\d(_)\d\d(_TIs_)\d\d(_TEs)', 'once'));
         if bTimeEncodedFME
             startDetails = regexp(jsonOut.SeriesDescription,'\d\d(_)\d\d(_TIs_)\d\d(_TEs)', 'once');
-            jsonOut.HadamardType = xASL_str2num(jsonOut.SeriesDescription(startDetails:startDetails+1));
-            jsonOut.HadamardNumberPLD = xASL_str2num(jsonOut.SeriesDescription(startDetails+3:startDetails+4));
-            jsonOut.HadamardNumberTE = xASL_str2num(jsonOut.SeriesDescription(startDetails+10:startDetails+11));
+            jsonOut.TimeEncodedMatrixSize = xASL_str2num(jsonOut.SeriesDescription(startDetails:startDetails+1));
+            jsonOut.TimeEncodedNumberPLD = xASL_str2num(jsonOut.SeriesDescription(startDetails+3:startDetails+4));
+            jsonOut.TimeEncodedNumberTE = xASL_str2num(jsonOut.SeriesDescription(startDetails+10:startDetails+11));
             fprintf('FME sequence, Hadamard-%d encoded images, %d PLDs, %d TEs\n', jsonOut.HadamardType, jsonOut.HadamardNumberPLD, jsonOut.HadamardNumberTE);
         end
     else
         bTimeEncodedFME = false;
     end
+    
+    % Check total acquired pairs for time encoded sequences
+    if ~isfield(jsonOut,'TotalAcquiredPairs')
+        
+        % Hadamard-4: 1 control / 3 label = 1 pair
+        % Hadamard-8: 1 control / 7 label = 1 pair
+        % etc.
+        
+        % Determine number of acquired pairs from image size and number of TE/TI
+        jsonOut.TotalAcquiredPairs = '?';
+        
+    end
```

Comment:
@BeatrizPadrela: I think it's your task to make this work as well 😉

Reply:
I agree :) Thanks for all the help